core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1531,9 +1540,9 @@ private[spark] class DAGScheduler(
             // Mark all the map as broken in the map stage, to ensure retry all the tasks on
             // resubmitted stage attempt.
             mapOutputTracker.unregisterAllMapOutput(shuffleId)
-          } else if (mapId != -1) {
+          } else if (mapIndex != -1) {
             // Mark the map whose fetch failed as broken in the map stage
-            mapOutputTracker.unregisterMapOutput(shuffleId, mapId, bmAddress)
+            mapOutputTracker.unregisterMapOutput(shuffleId, mapIndex, bmAddress)
           }
```

Comment:
the implicit `else` here is only if SHUFFLE_USE_OLD_FETCH_PROTOCOL, right?  worth at least a comment (otherwise that -1 will look more and more strange as no one uses the old shuffle protocol anymore).

Reply:
Great thanks for this comment, it's my mistake to use -1 as the default value of map index for old fetch protocol.
The logic of `mapId == -1` should only for `MetadataFetchFailedException`, it involoved in https://github.com/apache/spark/commit/7d71b9a56a4d644ccabb56dd282e84e2a49ef144 and #1227.
For our logic, just pass map index for both old and new protocol.